src/mesh/BoundingBox.cpp

```diff
@@ -108,58 +113,68 @@ int BoundingBox::getDimension() const
 
 const std::vector<double> BoundingBox::dataVector() const
 {
-  return _bounds;
+  std::vector<double> bounds;
+  if (_dimensions == 3) {
+    bounds.insert(bounds.end(), {_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1], _boundMin[2], _boundMax[2]});
+  } else {
+    bounds.insert(bounds.end(), {_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1]});
+  }
```

Comment:
Would this work as well, to reduce duplication?

```suggestion
  bounds.insert(bounds.end(), {_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1]});
  if (_dimensions == 3) {
    bounds.insert(bounds.end(), {_boundMin[2], _boundMax[2]});
  }
```

(why not `.push_back`?)

Reply:
Agree

Reply:
We (@jeremiahbintang ) decide to use `insert` instead of `push_back` since it's faster to add elements for `vector` (https://quick-bench.com/q/vqA7scnhikRpd8aHdzy_NtqJ3ig).